src/main/java/DeviceCodeFlow.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import com.microsoft.aad.msal4j.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+public class DeviceCodeFlow {
+
+    final static String PUBLIC_CLIENT_ID = "de165c80-4281-47c5-b23b-42ec6080db52";
```

Comment:
Do we want to have a client id hard coded here? We can just use placeholder text.

Reply:
Yes. Committed with client id by mistake, will remove it.